reV/SAM/generation.py

```diff
@@ -624,6 +625,256 @@ def default(self):
         return self._default
 
 
+class SolarThermal(Solar):
+    """ Base class for solar thermal """
+    def __init__(self, resource=None, meta=None, parameters=None,
+                 output_request=None):
+        """Initialize a SAM solar water heating object.
+
+        """
+        self.timezone = None
+        super().__init__(resource=resource, meta=meta, parameters=parameters,
+                         output_request=output_request)
+
+    def set_nsrdb(self, resource):
+        """
+        Set NSRDB resource file. Overloads Solar.set_nsrdb(). Solar thermal
+        PySAM models require a data file, not raw data.
+
+        Parameters
+        ----------
+        resource : pd.DataFrame
+            2D table with resource data. Available columns must have var_list.
+        """
+        self.time_interval = self.get_time_interval(resource.index.values)
+        # Create solar resource file
+        self._pysam_w_fname = self._create_pysam_wfile(self._meta, resource)
+        # pylint: disable=E1101
+        self[self._pysam_weather_tag] = self._pysam_w_fname
+
+    def _create_pysam_wfile(self, meta, resource, drop_leap=True):
+        """
+        Create PySAM weather input file.
+
+        Parameters
+        ----------
+        meta : pd.DataFrame
+            1D table with resource meta data.
+        resource : pd.DataFrame
+            2D table with resource data. Available columns must have var_list.
+        drop_leap : bool
+            Drops February 29th from the resource data. If False, December
+            31st is dropped from leap years.
+        """
+        fname = f'{self._site}_weather.csv'
```

Comment:
consider using .format() instead of format strings to be consistent with the rest of reV.

Reply:
Done